### PR TITLE
COIOS 593: ThreeDS Opaque error handling

### PR DESCRIPTION
**Pull request by robertdalmeida**
## Summary
<new> Errors from the 3ds sdk reported to the adyen backend  </new>'
**Why:** 
Adyen doesn't have visibility of the errors that merchant face during a 3ds challenge, to help the merchant understand errors from the sdk.

**What:**
We provide some error information back to the merchant as a base64 string.  This is passed back to the backend, so that adyen has visibility of the error that occurs and handle it accordingly.

## Pending
- [x] Update failing tests
- [x] Test with a live backend.

## Tested scenarios
- [x] UL testing of the sdk
- [x] Verification on a live backend some error scenarios. 
- [x] Sanity test of the checkout test app - sessions, dropin, component.

### CODE REVIEW

File: AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift
Comment (descorp) [inline AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift]: Should we wait for 3DS release?

Comment (robertdalmeida) [reply AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift]: Yes we should wait before the merge, the opaque error object is available only in the next release.

File: Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift
Comment (goergisn) [inline Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift]: Is this intended to be commented out / sample code?

**Review by goergisn — DISMISSED** [review]

File: Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift
Comment (robertdalmeida) [reply Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift]: Thanks, yes. Not intended to be there. Will remove it.

File: AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift
Comment (erenbesel) [inline AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift:192]: A weird one eh :) Let's create a private constant enum in this class and use that instead.

File: AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift
Comment (erenbesel) [inline AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift]: Many files have this without any change. You can revert these to reduce number of file changes

File: Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift
Comment (erenbesel) [inline Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift]: This way, the function captures the `paymentResponseHandler` strongly. 
We would need to send it as weak in a block which would be similar to the original way :)

File: Demo/Common/IntegrationExamples/DropIn/DropInAdvancedFlowExample.swift
Comment (erenbesel) [inline Demo/Common/IntegrationExamples/DropIn/DropInAdvancedFlowExample.swift]: Same here

File: AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift
Comment (robertdalmeida) [reply AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift:192]: Yeah :) a limitation in the backend to process the errors.  Done - with the Constant enum.

File: Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift
Comment (robertdalmeida) [reply Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift]: Indeed, i don't need to make any changes in the test apps anymore. Thanks. Reverted.

**Review by goergisn — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Pull request merged**
